model/collection.py

```diff
@@ -624,31 +625,40 @@ def unresolved_catalog(self, _db, data_source_name, operation):
 
         return query
 
-    def works_updated_since(self, _db, timestamp):
-        """Finds all works in a collection's catalog that have been updated
-           since the timestamp. Used in the metadata wrangler.
+    def licensepools_with_works_updated_since(self, _db, timestamp):
+        """Finds all LicensePools in a collection's catalog whose Works' OPDS
+        entries have been updated since the timestamp. Used by the
+        metadata wrangler.
+
+        :param _db: A database connection,
+        :param timestamp: A datetime.timestamp object
 
-           :return: a Query that yields (Work, LicensePool,
-              Identifier) 3-tuples. This gives caller all the
-              information necessary to create full OPDS entries for
-              the works.
+        :return: a Query that yields LicensePools. The Work and
+           Identifier associated with each LicensePool have been
+           pre-loaded, giving the caller all the information
+           necessary to create full OPDS entries for the works.
         """
         opds_operation = WorkCoverageRecord.GENERATE_OPDS_OPERATION
         qu = _db.query(
-            Work, LicensePool, Identifier
+            LicensePool
+        ).join(
+            LicensePool.work,
+        ).join(
+            LicensePool.identifier,
```

Comment:
You mentioned that there are two joins against the licensepools table but there should only be one. Are these different since they are joining against certain properties from LicensePool?